external/fv3fit/fv3fit/emulation/transforms/factories.py

```diff
@@ -142,6 +173,22 @@ def backward_names(self, requested_names: Set[str]) -> Set[str]:
             requested_names = factory.backward_names(requested_names)
         return requested_names
 
+    def backward_input_names(self) -> Set[str]:
```

Comment:
Little worried that this is doubling down on an implementation that's in the "debt" column.

Reply:
Hmm. Oli had to work around the inability to specify inputs for the backward transforms too, so I think it is a flexibility that we need as long as we want to allow transforms to advertise their own inputs/outputs. For the typical transform, these methods are simpler than the "backwards_names" and simply list ins/outs.

Thinking ahead, I think the path forward to pay down the debt is to separate the backwards and forwards transforms into separate objects with this signature `transform.{__call__/output_names/input_names}`. This current design would be easy to refactor to that setup simply by deleting the "backward_".

Reply:
Yeah, I think I agree with this path forward if I understand the change proposed.  Essentially, we have two separate transform groups.  One before model, one after model, and there's only a single "direction".  This would've solved my problem from before the v4 dataset switch as well.   Thanks for the extra context too, that Oli has had issues with this too.